tests/vmi_multus_test.go

```diff
@@ -375,32 +375,37 @@ var _ = Describe("[Serial]Multus", func() {
-			networks := []v1.Network{linuxBridgeNetwork}
-			linuxBridgeIfIdx := 0
 			customMacAddress := "50:00:00:00:90:0d"
 
 			It("[test_id:676]should configure valid custom MAC address on Linux bridge CNI interface.", func() {
 				By("Creating a VM with Linux bridge CNI network interface and default MAC address.")
-				vmiTwo := createVMIOnNode(interfaces, networks)
-				tests.WaitUntilVMIReady(vmiTwo, console.LoginToAlpine)
+				vmiTwo := libvmi.NewFedora(
+					libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()),
+					libvmi.WithNetwork(v1.DefaultPodNetwork()),
+					libvmi.WithInterface(linuxBridgeInterface),
+					libvmi.WithNetwork(&linuxBridgeNetwork),
+					libvmi.WithCloudInitNoCloudUserData(tests.GetGuestAgentUserData(), false),
+					libvmi.WithCloudInitNoCloudNetworkData(cloudInitNetworkDataWithStaticIPsByDevice("eth1", "10.1.1.2/24"), false))
+				vmiTwo = tests.StartVmOnNode(vmiTwo, nodes.Items[0].Name)
 
 				By("Creating another VM with custom MAC address on its Linux bridge CNI interface.")
-				interfaces[linuxBridgeIfIdx].MacAddress = customMacAddress
-				vmiOne := createVMIOnNode(interfaces, networks)
-				tests.WaitUntilVMIReady(vmiOne, console.LoginToAlpine)
-
-				By("Configuring static IP address to the Linux bridge interface.")
-				Expect(configInterface(vmiOne, "eth0", "10.1.1.1/24")).To(Succeed())
-				Expect(configInterface(vmiTwo, "eth0", "10.1.1.2/24")).To(Succeed())
+				linuxBridgeInterfaceWithCustomMac := linuxBridgeInterface
+				linuxBridgeInterfaceWithCustomMac.MacAddress = customMacAddress
+				vmiOne := libvmi.NewFedora(
+					libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()),
+					libvmi.WithNetwork(v1.DefaultPodNetwork()),
+					libvmi.WithInterface(linuxBridgeInterfaceWithCustomMac),
+					libvmi.WithNetwork(&linuxBridgeNetwork),
+					libvmi.WithCloudInitNoCloudUserData(tests.GetGuestAgentUserData(), false),
+					libvmi.WithCloudInitNoCloudNetworkData(cloudInitNetworkDataWithStaticIPsByMac(customMacAddress, "10.1.1.1/24"), false))
+				vmiOne = tests.StartVmOnNode(vmiOne, nodes.Items[0].Name)
+
+				tests.WaitUntilVMIReady(vmiOne, console.LoginToFedora)
+				tests.WaitUntilVMIReady(vmiTwo, console.LoginToFedora)
 
 				By("Verifying the desired custom MAC is the one that were actually configured on the interface.")
-				ipLinkShow := fmt.Sprintf("ip link show eth0 | grep -i \"%s\" | wc -l\n", customMacAddress)
-				err = console.SafeExpectBatch(vmiOne, []expect.Batcher{
-					&expect.BSnd{S: ipLinkShow},
-					&expect.BExp{R: "1"},
-				}, 15)
-				Expect(err).ToNot(HaveOccurred())
+				Expect(vmiOne.Status.Interfaces[1].MAC).To(Equal(customMacAddress), "should set linux bridge interface with the custom MAC address at VMI Status")
+				Expect(console.RunCommand(vmiOne, fmt.Sprintf("ip link show eth0 | grep -i \"%s\" | wc -l\n", customMacAddress), 15)).To(Succeed(), "should should set linux bridge interface with the custom MAC address at VMI")
```

Comment:
i dont think we should use the command, i would suggest please to do something as Alon did, poll the VMI status until it has the info.
Using the feature as VMI status, will give e2e coverage and would allow us to find problems.
"Low level" access skip some e2e testing imo

Reply:
I have already this discussion offline with @AlonaKaplan mac address is not taken from guest agent so we have to be sure that the guest is having the mac.

Reply:
See here please
https://github.com/kubevirt/kubevirt/pull/4448#discussion_r529412328
Mac is reported by guest agent, with all the derivations which are:
We should check it (even if not on this PR, we already check for example on test_id:3985).
Its prone for a race between test and Ga.
If we check, we should not use console imo, but just the VMI status, as this is the feature under testing.

Reply:
I have to find the source code but looks like we are ignoring MAC from guest agent and setting the one at libvirt domxml at least for linux bridge CNI for SRIOV it may be different.

Reply:
This is incorrect for this setup, please review the production code to confirm the answer quique provided earlier.

Reply:
> I have to find the source code but looks like we are ignoring MAC from guest agent and setting the one at libvirt domxml at least for linux bridge CNI for SRIOV it may be different, but still I am going to wait for guest agent and use @alonSadan helpers.

Ok thanks, i understand (since it takes from `domain.Spec.Devices.Interfaces`)
Btw, not part of this PR:
Ignoring the MAC from guest agent maybe should have an offline discussion if this is the right behavior,
as it looks on the Spec, without verifying actual GA, and update the VMI status accordingly as you said.